### PR TITLE
MINOR: Reduce ZK reads and ensure ZK watch is set for listener update

### PR DESCRIPTION
**Pull request by rajinisivaram**
Ensures that ZK watch is set for each live broker for listener update notifications in the controller by invoking `pathExists` since a watch is registered only if `exists` or `getData` is invoked after `registerZNodeChangeHandler`. Also avoids reading all brokers from ZooKeeper when a broker metadata is modified by passing in brokerId to `BrokerModifications` and reading only the updated broker.

The existing listener update test verifies both these changes. Earlier, the test did not detect missing watch for the last broker since metadata of all brokers were read from ZK (adding a watch for all) when any broker was updated.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by rajinisivaram** [comment]
@hachikuji @junrao This makes the change to read only the modified broker in ZK that @junrao suggested in PR #4488 . The change uncovered a bug in the code where a ZK watch was not being set for the last broker (this didn't trigger test failures earlier since we were reading all brokers from ZK). Can you review please? Thank you!

**Comment by ijuma** [comment]
@rajinisivaram, thanks for the PR. Can you please include the note about why this wasn't a problem before (we were rereading all brokers from zk) in the PR description?

**Comment by rajinisivaram** [comment]
@ijuma Added a note to the description, thanks.

File: core/src/main/scala/kafka/controller/KafkaController.scala
Comment (junrao) [inline core/src/main/scala/kafka/controller/KafkaController.scala]: We could just replace the above 2 statements with zkClient.registerZNodeChangeHandlerAndCheckExistence(brokerModificationsHandler).

File: core/src/main/scala/kafka/controller/KafkaController.scala
Comment (junrao) [inline core/src/main/scala/kafka/controller/KafkaController.scala]: We probably want to log $newMetadata.get?

**Review by junrao — APPROVED** [review]
@rajinisivaram : Thanks for the patch. LGTM. Just a couple of minor comments below. Once they are addressed, you can just merge it to trunk and probably 1.1.

**Comment by rajinisivaram** [comment]
@junrao Thanks for the review. I have made the updates. Will merge after builds complete.

**Comment by rajinisivaram** [comment]
Test failure unrelated, merging to trunk and 1.1.

**Pull request merged**
